hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,53 @@
+pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+fail_and_exit('Unknown pulp database name') unless pulp_db_param
+MONGO_DB_NAME = pulp_db_param.value
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  Dir.mkdir_p(export_dir)
```

Comment:
In Ruby 2.0, this is `FileUtils.mkdir_p`

Reply:
Switched.